arbitrary-users-patch/check_sidecar_image_digests.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+#
+# Copyright (c) 2018-2021 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+
+set -e
+
+SCRIPT_DIR=$(cd "$(dirname "$0")"; pwd)
+
+REGISTRY="quay.io"
+ORGANIZATION="eclipse"
+NAME_FORMAT="${REGISTRY}/${ORGANIZATION}"
+
+REGISTRY=${REGISTRY:-${DEFAULT_REGISTRY}}
+ORGANIZATION=${ORGANIZATION:-${DEFAULT_ORGANIZATION}}
+
+createPR() {
+    set +e
+    aBRANCH="$1"
+    PR_BRANCH="$2"
+
+    COMMIT_MSG="[update] Update digests in base_images"
+
+    # commit change into branch
+    git add "${SCRIPT_DIR}"/base_images
+    git commit -sm "${COMMIT_MSG}"
+
+    git branch "${PR_BRANCH}"
+    git checkout "${PR_BRANCH}"
+    git pull origin "${PR_BRANCH}"
+    git push origin "${PR_BRANCH}"
+    lastCommitComment="$(git log -1 --pretty=%B)"
+    hub pull-request -f -m "${lastCommitComment}" -b "${aBRANCH}" -h "${PR_BRANCH}"
+    set -e
+}
+
+BUILT_IMAGES=""
+while read -r line; do
+  dev_container_name=$(echo "$line" | tr -s ' ' | cut -f 1 -d ' ')
+  base_image_name=$(echo "$line" | tr -s ' ' | cut -f 2 -d ' ')
+  base_image_digest=$(echo "$line" | tr -s ' ' | cut -f 3 -d ' ' )
+  echo "Checking ${NAME_FORMAT}/${dev_container_name} based on $base_image_name ..."
+  latest_digest="$(skopeo inspect --tls-verify=false docker://"${base_image_name}" 2>/dev/null | jq -r '.Digest')"
```

Comment:
I think we can just use curl there for digest no ? (no need to have skopeo)

Reply:
HOWEVER... this is way more complicated as it requires doing some fancy footwork w/ docker authentication to be able to fetch manifests/digests. If the world was all quay, it'd be simple.

```
  # for quay, no need to authenticate
  AUTH=""
  # for docker, must authenticate with Bearer token
  if [[ ${base_image_name} == *"docker.io"* ]]; then
    REPOSITORY=${base_image_name#*/}; REPOSITORY=${REPOSITORY%:*} # get antora/antora
    TOKEN=$(curl -s "https://auth.docker.io/token?service=registry.docker.io&scope=repository:$REPOSITORY:pull" | jq -r .token)
    AUTH='-D - -H "Authorization: Bearer '$TOKEN'"'
    manifest_URL=${base_image_name/:/\/manifests\/}; manifest_URL=https://index.${manifest_URL/\//\/v2\/}
  else
    manifest_URL=${base_image_name/:/\/manifests\/}; manifest_URL=https://${manifest_URL/\//\/v2\/}
  fi
  manifest="$(curl -s ${AUTH} -H 'Accept: application/vnd.docker.distribution.manifest.v2+json' ${manifest_URL})"

  if [[ ${base_image_name} == "docker.io"* ]]; then
    latest_digest="sha256:$(echo $manifest | jq '.Docker-Content-Digest' -r)"
  else 
    latest_digest="sha256:$(echo $manifest | sha256sum | cut -f 1 -d ' ')"
  fi
  echo "latest digest ---> ${latest_digest}"
```

^^ that sort of works, but also doesn't :(

Reply:
yeah, I thought in this case it is probably simpler to defer complex logic to skopeo